clippy_lints/src/upper_case_acronyms.rs

```diff
@@ -56,55 +56,71 @@ impl UpperCaseAcronyms {
 
 impl_lint_pass!(UpperCaseAcronyms => [UPPER_CASE_ACRONYMS]);
 
-fn correct_ident(ident: &str) -> String {
-    let ident = ident.chars().rev().collect::<String>();
-    let fragments = ident
-        .split_inclusive(|x: char| !x.is_ascii_lowercase())
-        .rev()
-        .map(|x| x.chars().rev().collect::<String>());
-
-    let mut ident = fragments.clone().next().unwrap();
-    for (ref prev, ref curr) in fragments.tuple_windows() {
-        if <[&String; 2]>::from((prev, curr))
-            .iter()
-            .all(|s| s.len() == 1 && s.chars().next().unwrap().is_ascii_uppercase())
-        {
-            ident.push_str(&curr.to_ascii_lowercase());
+fn contains_acronym(s: &str) -> bool {
+    let mut count = 0;
+    for c in s.chars() {
+        if c.is_ascii_uppercase() {
+            count += 1;
+            if count == 3 {
```

Comment:
Why do we check for 3 here (but then 2 in the tail expression)? Doesn't this mean that we won't warn on `AB3` or `ABc`?
The documentation makes it sound like that we want to emit a warning there

> Triggers if there is more than one uppercase char next to each other

Reply:
Hmm, the old logic doesn't really seem to follow that very strictly either. The old logic does catch `AB3` but not `ABc`

Reply:
Because something like `AFoo` shouldn't lint, but `AS` should. It really shouldn't check the case for the third letter, but the old logic didn't handle that properly either.